tests/pipelines/test_pipelines.py

```diff
@@ -2014,3 +2015,40 @@ def test_ddpm_ddim_equality_batched(self):
 
         # the values aren't exactly equal, but the images look the same visually
         assert np.abs(ddpm_images - ddim_images).max() < 1e-1
+
+
+class TestLoraHotSwapping:
+    def test_hotswapping_peft_config_incompatible_raises(self):
+        # TODO
+        pass
+
+    def test_hotswapping_no_existing_adapter_raises(self):
+        # TODO
+        pass
+
+    def test_hotswapping_works(self):
+        # TODO
+        pass
+
+    def test_hotswapping_compiled_model_does_not_trigger_recompilation(self):
+        # TODO: kinda slow, should it get a slow marker?
+        env = {"TORCH_LOGS": "guards,recompiles"}
+        here = os.path.dirname(__file__)
+        file_name = os.path.join(here, "run_compiled_model_hotswap.py")
```

Comment:
To make the process faster, we could:

1. Use a small UNet with the config from https://github.com/huggingface/diffusers/blob/da18fbd54ce91b5ea5b9e01cb1677e4812a63203/tests/models/unets/test_models_unet_2d_condition.py#L378
2. Inject LoRAs into it with this config: https://github.com/huggingface/diffusers/blob/da18fbd54ce91b5ea5b9e01cb1677e4812a63203/tests/models/unets/test_models_unet_2d_condition.py#L67
3. Check for recompilation. 

If we decide to proceed this way, then we could do all of this inside the test case instead of having to do via a Python file under a capture logger context manager. But I am okay with having it done via a file as well.

Reply:
Could you please give me some guidance here? I tried to make the suggested changes (see new committed file) but `load_lora_weights` is not defined on the `unet`, so I would need to create a pipe still? Also, if I want to create the LoRA adapter in the test instead of loading an existing one, I would probably need to call `save_lora_weights` and store them in a temp folder, is that the right way?

> we could do all of this inside the test case instead of having to do via a Python file

That would be preferable but I'm not sure how we can capture the torch logs with `TORCH_LOGS=guards,recompiles` without shelling out. I don't think that something like `caplog` or `capsys` is sufficient, when I tried them, they were empty.

Reply:
Would something like this work?
https://github.com/huggingface/diffusers/blob/5d476f57c58c3cf7f39e764236c93c267fe83ca1/src/diffusers/loaders/lora_pipeline.py#L265

Reply:
I finally managed to get back to this topic. I tried to change the code according to the suggestion but ran into an issue.

So the adapter I was using so far, `"ybelkada/sd-1.5-pokemon-lora-peft"`, does not have the right shape for the small unet. Therefore, I'd have to create a new adapter that fits it but I'm not sure how. I can create a new adapter using `add_adapter` but there is no method to save this adapter, or is there? I can run `unet.save_pretrained` but that saves the whole model, not just the LoRA weights.

I tried saving just the LoRA weights using PEFT but when I use `unet.load_attn_procs` on that checkpoint, I get errors. I think it's because that method expected the state dict to already be conforming to the diffusers format, which is not the case.

Reply:
https://github.com/huggingface/diffusers/blob/a3e8d3f7deed140f57a28d82dd0b5d965bd0fb09/tests/lora/utils.py#L380 shows an example of saving the LoRA weights

https://github.com/huggingface/diffusers/blob/a3e8d3f7deed140f57a28d82dd0b5d965bd0fb09/tests/lora/utils.py#L386 loads it back

Does this help?

Reply:
To clarify, should I use a pipeline for this instead of using the unet directly or is there a way to use this with the unet and without a pipeline?

Reply:
For now, going via the pipeline route should be okay.

Reply:
Okay, I got it to work now. Thanks for the pointers.